utils/controller_debug.py

```diff
@@ -865,8 +866,11 @@ def GetResp(DbgPrint):
 
 #  timeout - How long (seconds) to wait for the response.  If
 #            not specified then a reasonable system default is used
+cmdLock = threading.Lock()
```

Comment:
Why do we, generally, need the lock here at all? Seems like we're not spawning any threads.

Reply:
I'm thinking of writing some more complex scripts which will run in their own threads.  This is in anticipation of needing to handle commands from multiple threads, but not necessary yet.

Reply:
I think it would be better to put off this part of this PR until there is a use case for it; in my experience speculative / "dead on arrival" code presents maintenance overhead but often ends up either dead forever if the anticipated use case never arrives, or rewritten because the anticipated use case ends up needing something quite different.

Reply:
The thing is, most of the scripts I write don't end up checked in to the repo because they aren't of general use.  Making the commands thread safe just allows me more flexibility in what I do in those scripts.
I'll pull it back out if you strongly object, but will likely just be adding it back in very soon

Reply:
I guess it's fine - but then please add some comments explaining why this is needed, so that somebody later doesn't look at this code, say "Hmm looks like it's not using any threads" and remove the locking. Even better would be to check in a couple of example scripts that need this, and link to them.